### DISCUSSION TITLE
Keybinding ignore exported var

### DISCUSSION THREAD

**Discussion by Tonus1** [i3/i3#4781] [Q&A]
Hi
I have a problem with libreoffice.
I want to launch it with a key (mod+shift+o) but can't have the correct UI :
From virtual terminal I pass before 
```
export SAL_USE_VCLPLUGIN=qt5
```
Problem is with key bindings or rofi it isn't taken and UI is ugly. 
Have I missed something ?

**Comment by Airblader** [comment]
> From virtual terminal

What does that mean? Where exactly do you set this?

i3 inherits the environment with which it is launched. If you need to pass this variable, either make sure it is set in i3's environment, or set it in your binding to execute the application.

**Reply by Tonus1** [reply]
When I use a key binding and when I launch xfce4-terminal, the following command do not show the same UI
```
export SAL_USE_VCLPLUGIN=gen && libreoffice
```
First is launched from xfce4-terminal

![2022-01-11_235434_screenshot](https://user-images.githubusercontent.com/11093251/149034464-1b201910-6922-4a77-b7d3-3041cbb0dd07.png)

Second is from `bindsym $mod+Shift+O exec --no-startup-id export SAL_USE_VCLPLUGIN=kf5 && libreoffice`

![2022-01-11_235348_screenshot](https://user-images.githubusercontent.com/11093251/149034462-7321e53f-ee34-4aa5-a467-51d4ca591c9a.png)

`export SAL_USE_VCLPLUGIN=kf5` should be set by `/etc/profile.d/libreoffice.sh`

For the record, this is Slackware-current (almost 15.0) with AlienBob's LibreOffice package.

**Reply by Airblader** [reply]
If you launch it from x4ce4-terminal without the environment variable, does it look like the one launched from the binding?

**Reply by Tonus1** [reply]
No, it looks like it should, as without the `export` part.
I believe it is because everything isn't sourced as it would be in a normal shell.
On my system
```
# /etc/profile: This file contains system-wide defaults used by
# all Bourne (and related) shells.
```

And from there is launched
```
# Append any additional sh scripts found in /etc/profile.d/:
for profile_script in /etc/profile.d/*.sh ; do
  if [ -x $profile_script ]; then
    . $profile_script
  fi
```

Where resides a libreoffice.sh which has the `export` command.

**Reply by Airblader** [reply]
/etc/profile isn't invoked for non-login shells. Essentially, `exec` can only provide the same environment with which i3 itself was started. See also [this](https://askubuntu.com/questions/247738/why-is-etc-profile-not-invoked-for-non-login-shells).

**Reply by Tonus1** [reply]
I tried to get this by modifying /etc/environment but no luck...
Tried to adapt with a script to invoque via the keybinding but no success either.
Do you have a clue in how I could manage to get that working ?
